src/main/java/org/opentripplanner/middleware/otp/OtpDispatcherResponse.java

```diff
@@ -90,4 +99,32 @@ public OtpDispatcherResponse clone() {
 //        clonedObject.responseBody = this.responseBody;
 //        return clonedObject;
     }
+
+    /**
+     * @return the first itinerary from the raw OTP response
+     * departing the same day as specified in the request date/time parameters, or null otherwise.
+     */
+    public Itinerary findItineraryDepartingSameDay(boolean tripIsArriveBy) {
+        OtpResponse response = this.getResponse();
+        TripPlan plan = response.plan;
+        HashMap<String, String> reqParams = response.requestParameters;
+        if (plan != null && plan.itineraries != null && reqParams != null) {
+            String requestDate = reqParams.get(DATE_PARAM);
+            String requestTime = reqParams.get(TIME_PARAM);
+            if (requestDate != null && requestTime != null) {
+                // Get the zone id for this plan. Don't search for itineraries in this plan if the zone cannot be determined.
+                Optional<ZoneId> fromZoneId = getZoneIdForCoordinates(plan.from.lat, plan.from.lon);
```

Comment:
I removed this ZoneId lookup in #75 as it turns out that this is not how OTP deals with timezones. Instead you'll want to use `DateTimeUtils. getOtpZoneId`.